test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -1478,6 +1478,28 @@ BOOST_AUTO_TEST_CASE(multi_level_mapping)
 	testContractAgainstCpp("f(uint256,uint256,uint256)", f, u256(5), u256(4), u256(0));
 }
 
+BOOST_AUTO_TEST_CASE(mapping_local_assignment)
+{
+	char const* sourceCode = R"(
+		contract test {
+			mapping(uint8 => uint8) m1;
+			mapping(uint8 => uint8) m2;
+			function f() public returns (uint8, uint8, uint8, uint8) {
+				mapping(uint8 => uint8) storage m = m1;
+				m[1] = 42;
+
+				m = m2;
```

Comment:
Can you add another end to end test that uses tuple assignment?

Reply:
Oh and please also check that `(m = m2)[1]=22` behaves correctly.

Reply:
See new tests below.